### PR TITLE
(fix) Updating glint-service to use the exported `URI` method from vscode-uri

### PR DESCRIPTION
**Pull request by robostheimer**
## What
Currently when any rehearsal command is used, the following error is emitted:
```       ^^^^^^^^^
SyntaxError: The requested module 'vscode-uri' does not provide an export named 'default'
```

There appears to have been api change in vscode-uri that rehearsal picked up as part of a package update.  This diff fixes the issue by using the exported URI function from vscode-uri.

Note, error will only occur if consuming app is on vscode-uri 3.0.8.  However, we should keep up with the package upgrade and ensure that no consumers are blocked by the above issue.

Note, we also need to bump glint/core dependency which contains the same issue being fixed in this diff: https://github.com/typed-ember/glint/blame/main/packages/core/src/language-server/util/index.ts

## Testing
- `pnmp vitest -u` passes when run on `packages/service` directory
```
✓ test/rehearsal-service.test.ts (6) 2430ms
   ✓ Test service (6) 2430ms
     ✓ construct
     ✓ getFileText
     ✓ setFileText
     ✓ setFileText
     ✓ getSourceFile 1542ms
     ✓ getDiagnostics 878ms

 Test Files  1 passed (1)
      Tests  6 passed (6)
   Start at  09:06:31
   Duration  6.79s (transform 3.06s, setup 0ms, collect 3.77s, tests 2.43s, environment 0ms, prepare 204ms)
```
- manual testing shows that the fix works and rehearsal commands can be run successfully with this diff.

### CODE REVIEW

File: packages/service/src/glint-service.ts
Comment (mikesherov) [inline packages/service/src/glint-service.ts:3]: Is this disabling comment needed anymore? I think that's specifically what 3.0.8 fixes.

Comment (robostheimer) [reply packages/service/src/glint-service.ts:3]: You are correct.  Will remove.

**Review by mikesherov — APPROVED** [review]
LGTM, will wait for @rwjblue to review before merging. Thank you!

File: packages/plugins/package.json
Comment (rwjblue) [inline packages/plugins/package.json:61]: Generally speaking bumping the minimum of a package in peer dependencies is a breaking change. However, for this specific case I searched GitHub (OSS) and internal repos for reference to `@rehearsal/plugins` (outside of lockfiles), and only found `@rehearsal/migrate` (which is also updated in this PR). So this is practically speaking going to be fine.

**Review by rwjblue — APPROVED** [review]

File: packages/plugins/package.json
Comment (rwjblue) [inline packages/plugins/package.json:61]: Generally speaking bumping the minimum of a package in peer dependencies is a breaking change. However, for this specific case I searched GitHub (OSS) and internal repos for reference to @rehearsal/plugins (outside of lockfiles), and only found @rehearsal/migrate (which is also updated in this PR). So this is practically speaking going to be fine.

Comment (robostheimer) [reply packages/plugins/package.json:61]: Ah, gotcha.  I was actually surprised that vscode-uri's change wasn't actually a major change.  They changed the api which caused consuming apps like rehearsal to mis-behave.

**Pull request merged**
